lib/commit-validator.js

```diff
@@ -1,31 +1,56 @@
+const lint = require('@commitlint/lint').default;
+
 class CommitValidator {
   constructor(options) {
-    this.patterns = [
-      new RegExp(
-        `^(${options.tasks.join('|')})\\((${options.scopes.join('|')})\\): [A-Z]{2,}-[0-9]+ .+`,
-      ),
-      new RegExp('^Revert ".+'),
-      new RegExp('^Releas(e|ing) v?\\d+\\.\\d+\\.\\d+'),
+    this.ignores = [
+      (commit) => new RegExp('^Releas(e|ing) v?\\d+\\.\\d+\\.\\d+').test(commit),
+      (commit) => new RegExp('^Revert ".+').test(commit),
+    ];
+
+    this.rules = {
+      'type-enum': [
+        2,
+        'always',
+        options.tasks
+          ? options.tasks
+          : ['feat', 'fix', 'docs', 'style', 'refactor', 'test', 'revert'],
+      ],
+      'scope-enum': [2, 'always', options.scopes ? options.scopes : []],
```

Comment:
In the same way, `scopes` should always be defined, let the parent provide always an array...